frontend/ui/widget/textboxwidget.lua

```diff
@@ -2186,4 +2183,228 @@ function TextBoxWidget:getSourceIndex(char_idx)
     end
 end
 
+function TextBoxWidget:getXtextHighlightIndices(start_x, start_y, end_x, end_y)
+    -- With xtext and fribidi, words may not be laid out in logical order,
+    -- so the left of a visual word may be its end in logical order,
+    -- and the right its start.
+    -- So, just find out charpos (text indice) of both points and
+    -- find word edges in the logical order text/charlist.
```

Comment:
No specific RTL/mixed BiDi stuff needed in textboxwidget.lua (unlike what you added in htmlboxwidget) ? It's messy, or it just all works fine because xtext does the right thing?
(When non-xtext, I guess we don't care as RTL/BiDi is not supported, that's why we introduced xtext.)

Reply:
With the BiDi text I tried the selection seemed to work correctly with xtext, so I did not add the code that I added to HtmlBoxWidget to prevent joining word rects.